### ISSUE TITLE
Batch Item Support

### ISSUE THREAD

**Issue by mikegrima** [Netflix/security_monkey#555]
Security Monkey's watching model generally works as follows:

1. List all items of a given technology and account from AWS
1. Describe all of the items in question
1. Check if any of those items are new or different.  If items are in the DB that are no longer present, then assume they are deleted.

_**Problems:**_
- The current paradigm struggles when working with very large quantities (order of thousands) of items.
- All of the item configurations obtained from AWS are stored in memory, which with larger workloads, becomes problematic

_**Desired state:**_
- SM should operate on batches of items at a time, as this will make very large workloads manageable

_**Current Progress:**_
I currently have a PR that is work in progress here: #502 .

- For technologies that are not region dependent:
  - Grab the full list of technologies
  - Describe a bunch of those items at a time
  - Send the batch to those items to the change finder
  - Repeat until complete.

- We still need to add regional technology dependencies
  - SOLUTION? Sync one region at a time?

_Current Problems:_
- The change finder will grab EVERYTHING that is in the DB for that given technology
- Because we have not yet fetched all the items from AWS, virtually everything will show up as deleted
  - This will also mark previously fetched items as deleted as well.

_Solution to current problems?_
- Before fetching anything from AWS, record the time <-- This will be the `sync_start_time`
- Refactor the current `find_changes()` such that deletions happen differently:
  - `find_changes(items_to_check, technology, account)`
  - `find_deleted(technology, account, sync_start_time)`
- The `find_changes` function will take the given list of items (batched), and check if there were any changes made to them
- ~~It will also log to the DB the current time that the item was last seen (`last_seen = now()`)~~
  - ~~This will require a new DB field added to the `Item` table to log this.~~
- ~~The `find_deleted` function will fetch a list of all items in a technology/account combo that has a `last_seen` date older than the `sync_start_time`, and mark them as deleted.~~
- The `find_deleted` function will use the following logic (as we implemented in Bananapeel):
```
    result = session.query(Item) \
        .join((Account, Item.account_id == Account.id)) \
        .join((Technology, Item.tech_id == Technology.id)) \
        .join((ItemRevision, Item.latest_revision_id == ItemRevision.id)) \
        .filter(Account.number == payload['account_number']) \
        .filter(Technology.name == payload['tech']) \
        .filter(ItemRevision.active == True) \
        .filter(Item.arn.notin_(arns)).all()
```

**Comment by mikegrima** [comment]
This is mostly implemented in #502

**Comment by mikegrima** [comment]
This is merged.
